src/index.ts

```diff
@@ -92,22 +97,38 @@ function setLocation() {
     if (points.length > 0)
         for (let i = -1; i < points.length - 1; i++) {
             const pt = points[(i + points.length) % points.length]; // Starting by the last point
-            pointsString += `&pt=${pt.x};${pt.y}`;
+            pointsString += `&pt=${pt.x};${pt.y}`;
         }
 
     const newRelativePathQuery = window.location.pathname + '?' + 'range=' + parameter + '&' + 'complexity=' + complexity + '&' + 'circles=' + Number(circles) + pointsString;
     history.pushState(null, '', newRelativePathQuery);
 }
 
 function initControls() {
-    parameterSlider.max = (fftSize - 1).toString();
-    parameter = parameterSlider.valueAsNumber;
+    const fftUnderSize = fftSize - 1, minParameter = Math.min(parameter, fftUnderSize), minComplexity = Math.min(complexity, fftUnderSize);
+    circles = complexityCircles.checked;
 
-    complexityNumber.max = (fftSize - 1).toString();
-    complexity = complexityNumber.valueAsNumber;
+    fft = new FFT(fftSize);
+    input = fft.createComplexArray();
+    output = fft.createComplexArray();
+    pathReinitialization();
+    rawPoints.forEach(pt => addPoint(pt.x, pt.y));
```

Comment:
Will this strategy work if you visit the page on your phone with no path initially present?

Reply:
The previous line being `fft = new FFT(fftSize);`, the fft will adapt at each try cycle, and the path should then be recomputed accordingly. Even on a phone with no path initially present.

Reply:
Ow, if `rawPoints`is null, you say?

Reply:
130ab27